sympy/polys/polytools.py

```diff
@@ -2669,6 +2675,128 @@ def subresultants(f, g):
 
         return list(map(per, result))
 
+    def subresultant_polys(f, g):
+        """
+        Computes the subresultant polynomials of two polynomials ``f`` and ``g``.
+        The returned list has length min(deg(f), deg(g))+1, and is in order of
+        increasing degree. The first polynomial in the list is the resultant.
+
+        This uses the subresultant PRS. The remainder r_i is the deg(r_{i-1})-th
+        subresultant polynomial. Additionally, if deg(r_i) < deg(r_{i-1}) - 1,
+        then the deg(r_i) subresultant polynomial is r_i * LC(r_i) ^ c_i, where
+        c_i = deg(r_{i-1})-deg(r_i)-1.
+
+        Examples
+        ========
+
+        >>> from sympy import Poly
+        >>> from sympy.abc import x
+
+        >>> Poly(x**2 + 1, x).subresultant_polys(Poly(x**2 - 1, x))
+        [Poly(4, x, domain='ZZ'),
+         Poly(-2, x, domain='ZZ'),
+         Poly(x**2 - 1, x, domain='ZZ')]
+
+        Observe that the first element is the resultant.
+
+        >>> Poly(x**2 + 1, x).resultant(Poly(x**2 - 1, x))
+        4
+
+        See also
+        ========
+
+        subresultants
+        subresultant_coeffs
+
+        """
+
+        # ensure deg(f) >= deg(g)
+        n = f.degree()
+        m = g.degree()
+
+        if n < m:
+            f, g = g, f
+            n, m = m, n
+
+        prs = f.subresultants(g)
+        if len(prs) <= 1:
+            return []
+
+        subres_polys = [Poly(0, f.gen) for _ in range(m + 1)]
+
+        subres_polys[-1] = g * g.LC() ** (n - m - 1)
+
+        for i in range(2, len(prs)):
+            curr = prs[i]
+            prev = prs[i-1]
+            curr_deg = curr.degree()
+            prev_deg = prev.degree()
+
+            # remainder r_i is the deg(r_{i-1})-th subres poly
+            subres_polys[ prev_deg -1 ] = curr
+
+            # if there is a "degree jump"
+            # if deg(r_i) < deg(r_{i-1}) - 1, then the deg(r_i) subres poly
+            # is r_i * LC(r_i) ^ c_i, where c_i = deg(r_{i-1})-deg(r_i)-1
+            degree_jump = prev_deg - curr_deg - 1
+            if degree_jump > 0:
+                subres_polys[curr_deg] = curr * curr.LC() ** degree_jump
```

Comment:
I was thinking like:
```python
zero = f.zero
subres_polys = []
deg = prs[-1].degree()

for p in reversed(prs[1:-1]):
    deg_new = p.degree()
    jump = deg_new - deg - 1
    if jump > 0:
        subres_polys.extend([zero] * jump)
        subres_polys[deg_new-1] = deg_new * p.LC() * jump**deg_new
    subres_polys.append(p)
    deg = deg_new
```
I'm sure this isn't quite right but that is sort of how I imagined the code would look.

Reply:
Thanks for the suggestion! I tried to incorporate the spirit of this. Basically, the list has to be long enough to assign the $deg(r_{i-1})$-th element. So, what I do now is append zeros until it is one short of that length, then append the required element (the remainder r_i). Then, the degree jump case is handled after in the loop, because it is guaranteed that the required index has already been created in the list.